File: Administrator.c
#include <stdio.h>
#include <stdlib.h>
#include <windows.h>

int IsAdministrator() {
    BOOL isAdmin = FALSE;
    PSID adminGroup;
    SID_IDENTIFIER_AUTHORITY NtAuthority = SECURITY_NT_AUTHORITY;

    if (AllocateAndInitializeSid(&NtAuthority, 2, SECURITY_BUILTIN_DOMAIN_RID, DOMAIN_ALIAS_RID_ADMINS,
                                 0, 0, 0, 0, 0, 0, &adminGroup)) {
        CheckTokenMembership(NULL, adminGroup, &isAdmin);
        FreeSid(adminGroup);
    }
    return isAdmin;
}

void RunAsAdministrator() {
    char exePath[MAX_PATH];
    GetModuleFileNameA(NULL, exePath, MAX_PATH);

    SHELLEXECUTEINFOA sei = { sizeof(sei) };
    sei.lpVerb = "runas";
    sei.lpFile = exePath;
    sei.nShow = SW_SHOWNORMAL;

    if (!ShellExecuteExA(&sei)) {
        printf("Failed to elevate privileges.\n");
    }
}

int main() {
    system("cls");

    if (!IsAdministrator()) {
        printf("Program requires Administrator privileges\n");
        RunAsAdministrator();
        return 0;
    }

    printf("Welcome I'm Mohammed Al-Baqer\n");

    try {
    } catch (...) {
        printf("Failed to execute the program.\n");
        system("pause");
    }

    return 0;
}

File: Administrator.cs
using System;
using System.Diagnostics;
using System.Security.Principal;
using System.Windows.Forms;

class Program {
    static bool IsAdministrator() {
        using (WindowsIdentity identity = WindowsIdentity.GetCurrent()) {
            WindowsPrincipal principal = new WindowsPrincipal(identity);
            return principal.IsInRole(WindowsBuiltInRole.Administrator);
        }
    }

    static void RunAsAdministrator() {
        ProcessStartInfo psi = new ProcessStartInfo {
            FileName = Application.ExecutablePath,
            Verb = "runas",
            UseShellExecute = true
        };
        try {
            Process.Start(psi);
        } catch {
            Console.WriteLine("Failed to elevate privileges.");
        }
    }

    [STAThread]
    static void Main() {
        Console.Clear();

        if (!IsAdministrator()) {
            Console.WriteLine("Program requires Administrator privileges");
            RunAsAdministrator();
            return;
        }

        Console.WriteLine("Welcome I'm Mohammed Al-Baqer");

        try {
        } catch (Exception e) {
            Console.WriteLine("Failed to execute the program: " + e.Message);
            Console.ReadLine();
        }
    }
}
